### PR TITLE
DB changes related to Project Transparency

### PR DESCRIPTION
**Pull request by MilanRusev**
* DB changes
* Ex_admin changes

### CODE REVIEW

File: lib/sanbase/db_scripts/import_ico_spreadsheet.ex [hunk 1]
```diff
@@ -18,17 +19,42 @@ defmodule Sanbase.DbScripts.ImportIcoSpreadsheet do
     ico_spreadsheet_row = ico_spreadsheet_row
     |> set_infrastructure_default()
 
-    project = fill_project(ico_spreadsheet_row)
+    project = insert_or_update_project(ico_spreadsheet_row)
+
+    ico = insert_or_update_ico(project, ico_spreadsheet_row)
+
+    insert_or_update_ico_currencies(ico, ico_spreadsheet_row)
```
Comment (IvanIvanoff) [inline lib/sanbase/db_scripts/import_ico_spreadsheet.ex]: These could be chained with the pipe operator |>

File: lib/sanbase/db_scripts/import_ico_spreadsheet.ex [hunk 2]
```diff
@@ -105,7 +131,8 @@ defmodule Sanbase.DbScripts.ImportIcoSpreadsheet do
 
   defp ensure_ico(project) do
     case project do
-      %{ico: result = %Ico{}} -> result
+      %{icos: []} -> %Ico{}
+      %{icos: result} when is_list(result) -> List.first(result)
```
Comment (IvanIvanoff) [inline lib/sanbase/db_scripts/import_ico_spreadsheet.ex]: better to use pattern matching `%{icos: [result|_]} -> result`

File: lib/sanbase/db_scripts/import_ico_spreadsheet.ex [hunk 3]
```diff
@@ -18,17 +19,40 @@ defmodule Sanbase.DbScripts.ImportIcoSpreadsheet do
...
+  defp insert_or_update_ico_currencies(ico, ico_spreadsheet_row) do
+    currencies = ensure_currencies(ico_spreadsheet_row.ico_currencies)
+    |> Enum.map(&Ecto.Changeset.change(&1))
+
+    Enum.each(currencies, fn currency ->
+      currency = currency
+      |> Repo.insert_or_update!()
+
+      ensure_ico_currency(ico, currency)
+      |> Repo.insert_or_update!()
```
Comment (valo) [inline lib/sanbase/db_scripts/import_ico_spreadsheet.ex]: Why do we need to have `Currency` and `IcoCurrency`? Can't we just have one model for a currency?

Comment (MilanRusev) [reply lib/sanbase/db_scripts/import_ico_spreadsheet.ex]: There is a many-to-many relationship between currencies and ICOs and IcoCurrency is the intermediate table. Only Currencies are used elsewhere too.

Comment (valo) [reply lib/sanbase/db_scripts/import_ico_spreadsheet.ex]: OK. We need to export our schema to a picture and start iterating on it. I have the feeling that we are complicating it too much :-)

**Pull request merged**